packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -677,6 +677,10 @@ function updateOffscreenComponent(
   const prevState: OffscreenState | null =
     current !== null ? current.memoizedState : null;
 
+  if (nextProps.mode === 'manual') {
+    markRef(current, workInProgress);
+  }
```

Comment:
What if it switches from `manual` to something else? We need to schedule a Ref effect to detach it. I think the test you wrote happens to work because it also schedules a Visibility effect during the same render.